### PR TITLE
Reactions

### LINKED ISSUE #175: Render emoji reactions
See https://github.com/zulip/zulip/issues/541

### PR DESCRIPTION
**Pull request by borisyankov**
* Recognize REACTION events and dispatch actions
* Render Reactions
* Aggregate reactions from raw data
* Handle state updates on reaction event updates
* Add reaction api calls. Handle reaction presses

Fixes #175

(based on `cache` PR)

### CODE REVIEW

**Comment by smarx** [comment]
*Automated message from Dropbox CLA bot*

@borisyankov, it looks like you've already signed the Dropbox CLA. Thanks!

**Comment by borisyankov** [comment]
Rebased against master.

**Comment by borisyankov** [comment]
Rebased against latest master.

File: src/chat/chatUpdater.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+export default (state, messageId, updater) => ({
```
Comment (neerajwahi) [inline src/chat/chatUpdater.js]: I like separating this out into a function!

**Pull request merged**